### ISSUE TITLE
Remove CATEGORICAL_ERROR cost from core NEAT-AI (code + tests)

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#2805]
## Summary
Hard-remove the non-standard `CATEGORICAL_ERROR` cost from core NEAT-AI — its class, registry entry, both descriptor-table entries, and every dedicated/incidental reference in source and tests — leaving lint, type-check, and the test suite green.

## Background
`CATEGORICAL_ERROR` is `1 − argmax accuracy`: a non-differentiable step function that is only meaningful as a **reported metric**, never as a training/selection cost. Policy: standard training costs live in the cost subsystem and downstream libraries; core must not carry a metric-only cost. The standard multi-class training cost — softmax + cross-entropy — already exists as `CROSS_ENTROPY`. Argmax/top-1 accuracy stays a *reporting* concern only.

This supersedes the abandoned surrogate-cost design from #2792 (now closed). No `surrogateCostName`, no worker-protocol changes.

> [!IMPORTANT]
> **Cross-repo sequencing.** Do NOT start this removal until `stSoftwareAU/NEAT-AI-Examples#523` (switch the MNIST example to `CROSS_ENTROPY`) has merged. Removing the cost first would leave that example referencing a deleted cost. This ordering cannot be enforced automatically across repos — confirm #523 is merged before picking this up.

## What Needs to Be Done

**Source**
- `src/costs/CategoricalError.ts` — **delete**.
- `src/Costs.ts` — drop the `CategoricalError` import, remove `CategoricalError.NAME` from `BUILT_IN_COST_NAMES`, and remove `this.registerFromInstance(new CategoricalError())`. (`BUILT_IN_COST_NAMES` goes 7 → 6; `BuiltInCostName` no longer includes `"CATEGORICAL_ERROR"`.)
- `src/costs/CostTaskDescriptor.ts` — remove the `{ costName: "CATEGORICAL_ERROR", … }` entry from `BUILT_IN_DESCRIPTORS`. Required: once `BuiltInCostName` loses the literal, the `satisfies TaskDescriptor[]` check fails otherwise.
- `src/costs/CostDescriptor.ts` — remove the `CATEGORICAL_ERROR` entry from `TABLE` and the matching row from the docstring table.
- `src/costs/CostAwareEarlyStop.ts` — update doc comments that cite `CATEGORICAL_ERROR` as an example to a surviving unit-range cost (e.g. `CROSS_ENTROPY`).
- `src/creature/CreatureTraining.ts` — update the three "continuous companion CE for CATEGORICAL_ERROR" seam comments to drop the reference.
- `src/config/RustScorerConfig.ts` — docstring: "seven `BUILT_IN_COST_NAMES`" → "six"; drop `CATEGORICAL_ERROR` from the listed names.

**Tests**
- `test/costs/CategoricalError.ts` — **delete**.
- `test/score/RustScorerCategoricalError.ts` — **delete**. The generic `--cost <NAME>` bridge plumbing is already covered by `test/score/RustScorerBridgeCostFlag.ts` (MAE/MSE/HINGE); this file only adds argmax-fixture specifics.
- `test/costs/CostName.ts` — remove `"CATEGORICAL_ERROR"` from the expected built-in name list.
- `test/costs/CostDescriptor.ts` — remove the `CATEGORICAL_ERROR maps to one_hot/unit/bounded_unipolar` test and the `isCostAware("CATEGORICAL_ERROR")` assertion.
- `test/costs/CostTaskDescriptor.ts` — remove the CATEGORICAL_ERROR descriptor-mapping test case(s).
- `test/costs/CostAwareEarlyStop.ts` — swap `CATEGORICAL_ERROR` → `CROSS_ENTROPY` (both unit-range; clamp/early-stop assertions unchanged in meaning).
- `test/costs/CostAwareEarlyStopIntegration.ts` — swap `CATEGORICAL_ERROR` → `CROSS_ENTROPY`.
- `test/config/TrainPerGen.ts` — swap `CATEGORICAL_ERROR` → a surviving supervised built-in (e.g. `CROSS_ENTROPY`); expected value stays `6` (`round(30 × 0.2)`).

## Acceptance Criteria
- [ ] `src/costs/CategoricalError.ts`, `test/costs/CategoricalError.ts`, and `test/score/RustScorerCategoricalError.ts` are deleted.
- [ ] `Costs.find("CATEGORICAL_ERROR")` throws `Unknown cost function`; `Costs.getAvailableCosts()` excludes it; `BUILT_IN_COST_NAMES.length === 6`.
- [ ] No file under `src/` or `test/` references `CATEGORICAL_ERROR` or `CategoricalError` (verify with a repo grep; `docs/` handled separately, `docs/archive/**` out of scope).
- [ ] Reported `error` tag, `targetError` early-stop, and champion-selection behaviour are unchanged for every surviving cost (re-targeted `CostAwareEarlyStop*` tests pass with identical meaning).
- [ ] `resolveDefaultTrainPerGen` still returns the supervised-scaled value for surviving built-in costs (TrainPerGen test green).
- [ ] `quality.sh` (lint, fmt, type-check, test) passes.

## Diagram
```mermaid
flowchart TD
    A["src/Costs.ts<br/>BUILT_IN_COST_NAMES − CATEGORICAL_ERROR"] -->|"BuiltInCostName type narrows"| B["src/costs/CostTaskDescriptor.ts<br/>satisfies check forces entry removal"]
    A --> C["delete src/costs/CategoricalError.ts"]
    C -->|"import removed"| D["delete test/costs/CategoricalError.ts<br/>delete test/score/RustScorerCategoricalError.ts"]
    A --> E["src/costs/CostDescriptor.ts<br/>remove TABLE entry"]
    A --> F["tests: CostName / CostDescriptor / CostTaskDescriptor<br/>drop CATEGORICAL_ERROR assertions"]
    A --> G["incidental swaps → CROSS_ENTROPY<br/>CostAwareEarlyStop*, TrainPerGen, code comments"]
```

## Dependencies
None in-repo. **Cross-repo prerequisite:** `stSoftwareAU/NEAT-AI-Examples#523` must merge first (see Background).

## Context
Part of #2798. Confirmed direction: remove the metric-only `CATEGORICAL_ERROR` from core; `CROSS_ENTROPY` is the standard multi-class training cost.

Note on size: ~16 files, but a single cohesive deletion. The touch points are coupled by the type checker (`BuiltInCostName` → `DescriptorCostName` `satisfies` check) and by test imports of the deleted class, so they must land together to keep the build green. The diff is deletion-dominated (one ~430-line redundant integration test plus the cost class and its unit test); it cannot be split smaller without leaving an intermediate red build.

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/2808 has been created to address this issue.

**Comment by stservice** [comment]
Closed automatically — PR has been merged.
